pwndbg/heap/__init__.py

```diff
@@ -1,12 +1,81 @@
+import pwndbg.color.message as message
+import pwndbg.config
 import pwndbg.heap.heap
 import pwndbg.symbol
 
 current = None
 
 heap_chain_limit = pwndbg.config.Parameter('heap-dereference-limit', 8, 'number of bins to dereference')
 
+resolve_via_heuristic = pwndbg.config.Parameter('resolve-via-heuristic', False,
+                                                'Resolve some missing symbols via heuristics')
+
+main_arena = pwndbg.config.Parameter('main_arena', "0", 'main_arena address for heuristics')
+
+thread_arena = pwndbg.config.Parameter('thread_arena', "0", 'thread_arena value for heuristics')
+
+mp_ = pwndbg.config.Parameter('mp_', "0", 'mp_ address for heuristics')
+
+tcache = pwndbg.config.Parameter('tcache', "0", 'tcache value for heuristics')
+
+global_max_fast = pwndbg.config.Parameter('global_max_fast', "0", 'global_max_fast address for heuristics')
+
+symbol_list = [main_arena, thread_arena, mp_, tcache, global_max_fast]
+
+
+@pwndbg.config.Trigger(symbol_list)
+def parse_config2address():
+    # Somehow, we can't use integer config because the integer is too big and will cause out of range error for GDB API
+    # So we convert the string to a int manually
+    for symbol in symbol_list:
+        if not isinstance(symbol.value, str):
+            continue
+        address_str = symbol.value.strip()
+        if address_str == "0":
+            continue
+        try:
+            if address_str.startswith("0b"):
+                address = int(address_str, 2)
+            elif address_str.startswith("0o"):
+                address = int(address_str, 8)
+            elif address_str.startswith("0x"):
+                address = int(address_str, 16)
```

Comment:
All those if/elif here can be replaced with `int(address_str, 0)` - it will determine the format

Reply:
Wow, I didn't know this cool feature before!
Thanks, I'll fix it!